### PR TITLE
Support node 16 and 18

### LINKED ISSUE #10: Create unit test and update docs once Node stream.pipeline Buffer-as-source bug is fixed
https://github.com/nodejs/node/issues/37731

### LINKED ISSUE #36: Update to support latest Node 16.7

### PR DESCRIPTION
**Pull request by carlansley**
Fixes #10
Fixes #36

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+      - name: Install dependencies
+        run: npm ci
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Audit Signatures
+        run: npm audit signatures
```
Comment (carlansley) [inline .github/workflows/ci.yml:30]: @adcreare note that we can use ^ this now with the latest version of node 16.  I think we should do this when building for deployment also.

File: src/node/socket.spec.ts [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
 /*
- * Copyright (c) 2021 Check Digit, LLC
+ * Copyright (c) 2021-2022 Check Digit, LLC
  *
  * This code is licensed under the MIT license (see LICENSE.txt for details).
  */
 
-import assert from 'assert';
-import net from 'net';
+import assert from 'node:assert';
```
Comment (ramaghanta) [inline src/node/socket.spec.ts]: `{ strict as assert }`

File: src/node/stream-consumers.spec.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
+// node/stream-consumers.spec.ts
+
+/*
+ * Copyright (c) 2021-2022 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import assert from 'node:assert';
```
Comment (ramaghanta) [inline src/node/stream-consumers.spec.ts]: `{ strict as assert }`

File: src/node/stream-consumers.spec.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import * as stream from 'node:stream';
```
Comment (ramaghanta) [inline src/node/stream-consumers.spec.ts]: `import stream from 'node:stream'`

**Review by ramaghanta — CHANGES_REQUESTED** [review]
couple of minor changes

**Review by adcreare — APPROVED** [review]
LGTM

**Review by ramaghanta — APPROVED** [review]
lgtm

**Pull request merged**